services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -2093,6 +2094,255 @@ Environment "acceptance-de" has both upstream.latest and upstream.environment co
+							Latest:      false,
+							Environment: "dev",
+						},
+					},
+				},
+				&DeployApplicationVersion{
+					Application: "test",
+					Environment: "dev",
+					Version:     1,
+				},
+			},
+			ReleaseTrainEnv: "staging",
+			shouldSucceed:   true,
+			expectedError:   "",
+			expectedCommitMsg: `Release Train to environment/environment group 'staging':
+
+Release Train to 'staging' environment for 1 apps:
+
+The release train deployed 1 services from 'dev' to 'staging'
+deployed version 1 of "test" to "staging"
+
+
+
+
+`,
+		},
+		{
+			Name: "Release train done with commit but nothing to deploy",
+			SetupTransformers: []Transformer{
+				&CreateEnvironment{
+					Environment: "dev",
+					Config:      testutil.MakeEnvConfigLatest(nil),
+				},
+				&CreateApplicationVersion{
+					Application: "test",
+					Version:     1,
+					Manifests: map[string]string{
+						"dev": "dev",
+					},
+				},
+				&DeployApplicationVersion{
+					Application: "test",
+					Environment: "dev",
+					Version:     1,
+				},
+			},
+			ReleaseTrainEnv: "dev",
+			shouldSucceed:   true,
+			expectedError:   "",
+			expectedCommitMsg: `Release Train to environment/environment group 'dev':
+
+Release Train to 'dev' environment for 0 apps:
+
+The release train deployed 0 services from 'latest' to 'dev'
+
+
+
+`,
+		},
+		{
+			Name: "Release train with invalid commitHash",
+			SetupTransformers: []Transformer{
+				&CreateEnvironment{
+					Environment: "dev",
+					Config:      testutil.MakeEnvConfigLatest(nil),
+				},
+				&CreateApplicationVersion{
+					Application: "test",
+					Version:     1,
+					Manifests: map[string]string{
+						"dev": "dev",
+					},
+				},
+				&DeployApplicationVersion{
+					Application: "test",
+					Environment: "dev",
+					Version:     1,
+				},
+			},
+			ReleaseTrainEnv:    "dev",
+			shouldSucceed:      false,
+			expectedError:      "rpc error: code = InvalidArgument desc = error: could not get app version for commitHash 3f1debc97f5880c59caab9b36ad31f52604ce4dd for dev: ErrNotFound: object not found - no match for id (3f1debc97f5880c59caab9b36ad31f52604ce4dd)",
+			overrideCommitHash: "3f1debc97f5880c59caab9b36ad31f52604ce4dd",
+		},
+		{
+			Name: "Release train with invalid oid value",
+			SetupTransformers: []Transformer{
+				&CreateEnvironment{
+					Environment: "dev",
+					Config:      testutil.MakeEnvConfigLatest(nil),
+				},
+				&CreateApplicationVersion{
+					Application: "test",
+					Version:     1,
+					Manifests: map[string]string{
+						"dev": "dev",
+					},
+				},
+				&DeployApplicationVersion{
+					Application: "test",
+					Environment: "dev",
+					Version:     1,
+				},
+			},
+			ReleaseTrainEnv:    "dev",
+			shouldSucceed:      false,
+			expectedError:      "rpc error: code = InvalidArgument desc = error: could not get app version for commitHash aa for dev: Error creating new oid for commitHash aa: invalid oid",
+			overrideCommitHash: "aa",
```

Comment:
I'd like to see 1 additional test where `overrideCommitHash!=""` and the release train is successful.